### ISSUE TITLE
InvalidOperationException "buffer must have a project entry before parsing"

### ISSUE THREAD

**Issue by huguesv** [Microsoft/PTVS#3315]
I started VS and had multiple REPL windows docked (hidden).
As I started closing them, it had the effect of making REPL windows visible, and I got this assert:
![image](https://user-images.githubusercontent.com/1696845/32623698-dcfed74e-c53b-11e7-97d1-212c59e99afa.png)

```
            var parser = entry.GetOrCreateBufferParser(_services);
            try {
                parser.AddBuffer(subjectBuffer);
            } catch (InvalidOperationException ex) {
                // Should not fail here due to missing analysis entry
                // But if we do, it probably means someone else owns
                // the buffer now and we should quietly let them.
                Debug.Fail(ex.ToUnhandledExceptionMessage(GetType()));
                return;
            }
```

It's one of those debug assert dialogs that you can't click on because it's either busy or it doesn't show up at all.

**Comment by huguesv** [comment]
Last seen today while applying django migrations.

**Comment by zooba** [comment]
This should be fixed in #3577 - there isn't actually anyone else connecting the subject buffer at this point (where the analyzer is disposed), so we have to do the retry immediately.

**Comment by huguesv** [comment]
Haven't seen this in a while.
